### ISSUE TITLE
Add "branch" + "project name" to window title

### ISSUE THREAD

**Issue by humdingerb** [Genio-The-Haiku-IDE/Genio#413]
The window title shows the file name, which at least if you show the complete path, is enough to know what project the current file belongs to. If you only show the file name with no path this might difficult.

Also important is the branch the file belongs to. More so, if your project has so many files that the project name and branch in the Projects pane got scrolled out of view.

Suggestion for the window title:

* if complete path is shown: "Genio: /path/to/current/file [branch]"
* if only filename is shown: "Genio: Project name - file name [branch]"

**Comment by Freaxed** [comment]
Is the "Genio:" prefix really needed?

**Comment by humdingerb** [comment]
No, you're right. The Genio window should be recognizable enough without it... :)

**Comment by jackburton79** [comment]
I tried to implement this here: https://github.com/Genio-The-Haiku-IDE/Genio/tree/feature/branchInTitle . The only issue is when you change branch, the current title is not updated unless you change the active file.

**Comment by jackburton79** [comment]
@humdingerb we implemented this partially. What do you think about the current situation ?

**Comment by humdingerb** [comment]
It appears that the "Show full path" option doesn't quite work. That is, when deactivated there's neither filename (without path), nor the branch name in the window title.

Also, there's always the "Genio" in front, which I think was considered not to be needed.

Maybe there should be an additonal option, "Show project name in window title". Then you'd have these window title options (with/without "project name" and "branch name"):

* Project name : File name [branch name]
* Project name : /path/to/File name [branch name]

**Comment by jackburton79** [comment]
> > It appears that the "Show full path" option doesn't quite work. That is, when deactivated there's neither filename (without path), nor the branch name in the window title.
> 
> Yeah, I noticed that too some time ago, I guess that's a regression ?

This is now fixed

**Comment by humdingerb** [comment]
> Ok, but what should be shown if there aren't any open file, then ? Nothing ?
> Or, if no file is open, we should show it ?
Makes sense, show "Genio" when no file is open.

**Comment by humdingerb** [comment]
Works nicely!
Now we just (optionally) need the project name up there; App.cpp and MainWindow.cpp are really quite common file names... :)

**Comment by humdingerb** [comment]
+1

**Comment by jackburton79** [comment]
Done!

**Comment by humdingerb** [comment]
Perfect! 👍
